sql/core/src/main/scala/org/apache/spark/sql/execution/python/WindowInPandasExec.scala

```diff
@@ -158,6 +158,8 @@ case class WindowInPandasExec(
   }
 
   protected override def doExecute(): RDD[InternalRow] = {
+    metrics // force lazy init at driver
```

Comment:
What happens if we don't do this force lazy init?

Reply:
If I do not force lazy init the test [sql/core/src/test/resources/sql-tests/inputs/udf/postgreSQL/udf-aggregates_part3.sql](https://github.com/apache/spark/pull/33559/files#diff-573927f4713989a687ff60ffc00dd3c9a941475be8e7e3bf206a251682238f74) fails, see discussion above.